astropy/units/tests/test_quantity_non_ufuncs.py

```diff
@@ -2243,7 +2243,6 @@ def test_slogdet(self):
         # TODO: Could be supported if we had a natural logarithm unit.
         with pytest.raises(TypeError):
             logdet = np.linalg.slogdet(self.q)
-            assert hasattr(logdet, "unit")
```

Comment:
Doesn't this removal change the test?

Reply:
Don't see how `hasattr` could ever raise `TypeError` on the first parameter, so it shouldn't – but `slodget` is returning a tuple, so that `assert` seems out of scope anyway.

Reply:
And if the `np.linalg.slogdet()` call raises an error then `logdet` isn't defined anyways. So the deletion of the line is indeed correct.